internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go

```diff
@@ -370,13 +442,92 @@ func flattenTKGSTopologyControlPlane(controlPlane *tkgservicevspheremodel.Vmware
 		return nil
 	}
 
+	vms := make([]interface{}, 0)
+
+	for _, vs := range controlPlane.Volumes {
+		vms = append(vms, flattenTKGSVolumes(vs))
+	}
+
+	flattenControlPlane[volumesKey] = vms
 	flattenControlPlane[classKey] = controlPlane.Class
 	flattenControlPlane[storageClassKey] = controlPlane.StorageClass
 	flattenControlPlane[highAvailabilityKey] = controlPlane.HighAvailability
 
 	return []interface{}{flattenControlPlane}
 }
 
+var tkgServiceVolumes = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Configurable volumes for control plane nodes",
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			capacityKey: {
+				Type:        schema.TypeFloat,
+				Description: "Volume capacity is in gib",
+				Optional:    true,
+			},
+			mountPathKey: {
+				Type:        schema.TypeString,
+				Description: "It is the directory where the volume device is to be mounted",
+				Optional:    true,
+			},
+			volumeNameKey: {
+				Type:        schema.TypeString,
+				Description: "It is the volume name",
+				Optional:    true,
+			},
+			pvcStorageClassKey: {
```

Comment:
How is this values populated? I don't see this in the UI. is it a computed value?

Reply:
It is the Storage Class value under Volumes in the UI